misk-hibernate/src/main/kotlin/misk/hibernate/SessionFactoryService.kt

```diff
@@ -160,7 +165,32 @@ internal class SessionFactoryService(
     } else if (BoxedStringType.isBoxedString(field.type.kotlin)) {
       value.typeName = BoxedStringType::class.java.name
       value.setTypeParameter("boxedStringField", field)
+    } else {
+      for (annotation : Annotation in field.annotations) {
```

Comment:
This reads like a search, can it be done without iteration (e.g., using a map type?)

Reply:
I'm not sure if it can since I need to check if any of those annotations have an annotation on them; i can't just query by type since they can be any type. I'll look into other mechanisms for getting that list however

Reply:
Still getting used to the GitHub UI (I'm used to Bitbucket), so maybe I missed the broader context. Is this happening once (e.g., at app startup), or at the rate of DB connections, or some other repeated frequency? If it's repeated, can the results of this lookup be memoized by field?

Reply:
This gets invoked when Hibernate is initialized.